client/my-sites/themes/theme-showcase.jsx

```diff
@@ -171,7 +222,8 @@ const ThemeShowcase = React.createClass( {
 } );
 
 export default connect(
-	state => ( {
+	( state, { siteId } ) => ( {
+		siteSlug: getSiteSlug( state, siteId ),
 		queryParams: getQueryParams( state ),
 		themesList: getThemesList( state ),
```

Comment:
`queryParams` and `themesList` aren't really used by `ThemeShowcase` itself anymore but merely passed on to `ThemesSelection`. So we should remove them here and move them instead to `ThemesSelection`'s `connect()` expression. (This, too, will allow us to encapsulate the results list more in `ThemesSelection`.)